.github/workflows/build-wasm-internal.yml

```diff
@@ -0,0 +1,76 @@
+---
+name: Build @bitwarden/sdk-internal
+
+on:
+  pull_request:
+  push:
+    branches:
+      - "main"
+      - "rc"
+      - "hotfix-rc"
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+    working-directory: crates/bitwarden-wasm-internal
+
+jobs:
+  build:
+    name: Building @bitwarden/sdk-wasm-internal
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Setup Node
+        uses: actions/setup-node@0a44ba7841725637a19e28fa30b79a866c81b0a6 # v4.0.4
+        with:
+          node-version: 20
+          registry-url: "https://npm.pkg.github.com"
+          cache: "npm"
+
+      - name: Install dependencies
+        run: npm i -g binaryen
+
+      - name: Install rust
+        uses: dtolnay/rust-toolchain@7b1c307e0dcbda6122208f10795a713336a9b35a # stable
+        with:
+          toolchain: stable
+          targets: wasm32-unknown-unknown
+
+      - name: Cache cargo registry
+        uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84 # v2.7.3
+        with:
+          key: wasm-cargo-cache
+
+      - name: Install wasm-bindgen-cli
+        run: cargo install wasm-bindgen-cli
+
+      - name: Build
+        run: ./build.sh -r
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@50769540e7f4bd5e21e526ee35c689e35e0d6874 # v4.4.0
+        with:
+          name: sdk-bitwarden-wasm-internal
+          path: ${{ github.workspace }}/languages/js/sdk-internal/*
+          if-no-files-found: error
+
+      - name: Set version
+        if: ${{ github.ref == 'refs/heads/main' }}
+        # Fetches current version from registry and uses prerelease to bump it
+        run: |
+          npm version --no-git-tag-version $(npm view @bitwarden/sdk-internal@latest version)
+          npm version --no-git-tag-version prerelease
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        working-directory: languages/js/sdk-internal
+
+      - name: Publish NPM
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        working-directory: languages/js/sdk-internal
```

Comment:
We should split the build and publish workflows. If it's needed to publish it on every push to `main`, I suggest triggering the publishing workflow here. An example of that can be found in the build-swift.yml workflow: https://github.com/bitwarden/sdk/blob/90f8a5461e5e7ed07b1f5a13fc2fb793994a6fb5/.github/workflows/build-swift.yml#L101-L115

BTW is it released to [npmjs.com](https://www.npmjs.com/) or to github packages?

Reply:
Currently gh packages, but we will probably change it to npmjs.